phases/11-llm-engineering/15-prompt-caching/code/main.py

```
@@ -0,0 +1,166 @@
+
+
+def main() -> None:
+    traffic = make_traffic(n_requests=500, n_prefixes=3, prefix_size=15000, suffix_size=400)
+    seconds_between = 4  # one request every 4 seconds
+
+    anthro_5m = simulate_anthropic(traffic, ttl_seconds=300, seconds_between=seconds_between)
+    anthro_1h = simulate_anthropic(traffic, ttl_seconds=3600, seconds_between=seconds_between)
+    openai = simulate_openai(traffic, seconds_between=seconds_between)
+    gemini = simulate_gemini(traffic, ttl_seconds=3600, seconds_between=seconds_between)
+
+    print(f"Scenario: 500 requests, 3 rotating prefixes (15K tok each), 4s apart\n")
```

Ruff (0.15.10)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)